### PR TITLE
Add ConfirmationModal and ConfirmationButton components [WD-4255]

### PR DESCRIPTION
**Pull request by lorumic**
## Done

- Added ConfirmationModal component
- Added ConfirmationButton component

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Check an usage example of the 2 new components in the Storybook

## Fixes

Fixes: [WD-4255](https://warthogs.atlassian.net/browse/WD-4255).

[WD-4255]: https://warthogs.atlassian.net/browse/WD-4255?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lorumic** [comment]
I have added more example use cases in the Storybook, both for the `ConfirmationModal` and for the `ConfirmationButton`.

File: src/components/ConfirmationButton/ConfirmationButton.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+### Base appearance
+
+<Canvas>
+  <Story name="Base appearance">
+    {() => {
+      const [isLoading, setLoading] = useState(false);
+      const brieflyLoad = () => {
+        setLoading(true);
+        setTimeout(() => setLoading(false), 2000);
```
Comment (huwshimi) [inline src/components/ConfirmationButton/ConfirmationButton.stories.mdx:106]: For some reason these states aren't doing anything for me.

**Review by huwshimi — COMMENTED** [review]
This is working nicely. I've made a suggestion about how the props could work that might mean a number of small changes so if you agree I'll do another review after you update it.

File: src/components/ConfirmationButton/ConfirmationButton.stories.mdx (hunk 1 above)
Comment (lorumic) [reply src/components/ConfirmationButton/ConfirmationButton.stories.mdx:106]: Really? Strange, this is what I see when I click on the buttons with the loading state:

Peek 2023-06-16 10-06

This state was added just to provide a quick example of a fake "loading" operation starting upon confirmation.

File: src/components/ConfirmationButton/ConfirmationButton.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            {showShiftClickHint && (
+              <p className="p-modal__shortcut-hint u-text--muted">
+                You can skip the confirmation dialog by holding{" "}
+                <code>SHIFT</code> when clicking on the action.
+              </p>
+            )}
+          </ConfirmationModal>
+        </Portal>
+      )}
+      <Button
```
[the remark this answers is not in the document]
Comment (lorumic) [reply src/components/ConfirmationButton/ConfirmationButton.tsx:108]: True, the `ActionButton` should allow us to encapsulate the loading logic much better. Thanks for the suggestion.

File: src/components/ConfirmationButton/ConfirmationButton.stories.mdx (hunk 1 above)
Comment (huwshimi) [reply src/components/ConfirmationButton/ConfirmationButton.stories.mdx:106]: Oh, ignore me, I thought the spinner was supposed to appear when you click the button to open the modal, but it actually appears when you confirm!

**Review by huwshimi — APPROVED** [review]
Thanks for those updates! This looks like it will work nicely for Juju Dashboard now too.

**Pull request merged**